mllib/src/main/scala/org/apache/spark/mllib/feature/ChiSqSelector.scala

```diff
@@ -197,3 +197,28 @@ class ChiSqSelector @Since("1.3.0") (
     new ChiSqSelectorModel(indices)
   }
 }
+
+/**
+ * Creates a ChiSquared feature selector by False Positive Rate (FPR) test.
+ * @param alpha the highest p-value for features to be kept
+ */
+@Since("2.1.0")
+class ChiSqSelectorByFpr @Since("2.1.0") (
+  @Since("2.1.0") val alpha: Double) extends Serializable {
+
+  /**
+   * Returns a ChiSquared feature selector by FPR.
+   *
+   * @param data an `RDD[LabeledPoint]` containing the labeled dataset with categorical features.
+   *             Real-valued features will be treated as categorical for each distinct value.
+   *             Apply feature discretizer before using this function.
+   */
+  @Since("2.1.0")
+  def fit(data: RDD[LabeledPoint]): ChiSqSelectorModel = {
+    val indices = Statistics.chiSqTest(data)
+      .zipWithIndex.filter { case (res, _) => res.pValue < alpha }
```

Comment:
Two or more models based on the same chi-squared test is reasonable, because the chi-squared test results contain "pValue,  degreesOfFreedom,  statistic,,,". numTopFeatures selection method uses statistic, and FPR selection method uses pValue. 

Expose the p-value to the caller maybe useful, but we can expose ChiSqTestResult, not only p-value.